### PR TITLE
22495 update furnishing document endpoint to support stage 1 transition letter

### PR DESCRIPTION
**Pull request by kzdev420**
*Issue #:* /bcgov/entity#22495

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

**Comment by kzdev420** [comment]
NO_AR
Screenshot 2024-08-19 at 3 08 29 PM

NO_AR_EP
Screenshot 2024-08-19 at 3 10 55 PM

NO_TR
Screenshot 2024-08-19 at 3 11 57 PM

File: legal-api/tests/unit/models/__init__.py [hunk 1]
```diff
@@ -459,7 +459,7 @@ def factory_furnishing(business_id,
     return furnishing
 
 
-def factory_business_with_stage_one_furnishing(legal_type=None):
+def factory_business_with_stage_one_furnishing(legal_type=None, furnishing_name=None):
```
Comment (chenhongjing) [inline legal-api/tests/unit/models/__init__.py:462]: Can you fix the broken unit tests due to this change?

File: legal-api/report-templates/noticeOfDissolutionCommencement.html [hunk 2]
```diff
@@ -31,14 +31,24 @@
...
+            <div class="mt-5">If within one month after the date of this letter, the company fails to file the outstanding annual reports, a notice may be published on the BC Laws website <a href="https://www.bclaws.gov.bc.ca/" class="break-url">www.bclaws.ca</a>. This notice will state that, at any time after the expiration of one month after the date of publication of the notice, the company will be dissolved, unless cause is shown to the contrary; I am satisfied the failure has been or is being remedied; or a copy of the entered court order to the contrary has been filed.</div>
+          {% endif %}
+        {% else %}
+            <div class="mt-5">Under section 422 of the Business Corporation Act (the Act), this letter is to notify you that your company has for two years failed to file the PRTA required under section 51 of the Act. A company must annually, within two months after each anniversary of the date on which the company was recognized, file a PRTA with the Registrar.</div>
```
Comment (chenhongjing) [inline legal-api/report-templates/noticeOfDissolutionCommencement.html:50]: Can you confirm with Dylan about this part of wording for PRTA? It's not filed annually.

**Comment by dylan644** [comment]
1st letter - corps with overdue ARs is good
2nd letter - EP with overdue ARs is good
 
3rd letter - corp with overdue PRTA needs some changes - that is our fault, when Andy and I were putting together the emails we initially missed the letters as separate items and then in a rush we just put the first two together as full examples and we hoped whoever worked on this would just figure it out, but we were wrong, more customization is need for that letter.
 
Leave the third one with me, i'll create a new ticket to enact the changes to that

**Review by chenhongjing — APPROVED** [review]
Approve. As Dylan said, we'll correct the wordings in letter in another ticket.

**Pull request merged**